.github/workflows/update-and-publish-helm-chart.yml

```
@@ -57,3 +62,109 @@ jobs:
+                                    # delete
+                                    #   (1) everything that isn't between these two strings
+                                    #   (2) each line's leading whitespace
+            tail -1 |               # get the last line of the resulting text.
+            awk -F'"' '{print $2}'  # extract the version number from between the quotes.
+          }
+
+          function getVersionFromDocker(){
+            # A function for extracting the current chart version from Leonardo's Dockerfile
+            grep "ENV CROMWELL_CHART_VERSION" $1 | # search for the line where the chart version is set.
+            awk -F'[ ]' '{ print $3 }'             # print the third whitespace-separated word (the version).
+          }
+
+          cd leonardo
+
+          versionFromReference=$(getVersionFromReference http/src/main/resources/reference.conf)
+          versionFromDocker=$(getVersionFromDocker Dockerfile)
+
+          if [ $versionFromReference != $versionFromDocker ]; then
+            echo "ERROR: Cromwell versions in reference.conf and Dockerfile do not match." \
+                 "($versionFromReference != $versionFromDocker)" \
+            exit 1
+          fi
+
+          OLD_CROMWELL_HELM_V=$versionFromReference
+
+          NEW_BRANCH_NAME="cromwell_helm_${NEW_CROMWELL_HELM_V}-AUTOMERGE"
+          TEMP=$(mktemp)
+          git fetch
+          git checkout -b "${NEW_BRANCH_NAME}"
+
+          SED_STRING="s/${versionFromReference}/${NEW_CROMWELL_HELM_V}/g"
+          FILES=( "Dockerfile" "http/src/main/resources/reference.conf" "http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/ConfigReaderSpec.scala" "http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/KubernetesTestData.scala" )
+          for i in "${FILES[@]}"
+          do
+            echo "Updating ${i}"
+            sed "${SED_STRING}" ${i} > ${TEMP}
+            mv ${TEMP} ${i}
+            git add ${i}
+          done
+
+          echo "Pushing to leonardo branch ${NEW_BRANCH_NAME}"
+          git config --global user.name "broadbot"
+          git config --global user.email "broadbot@broadinstitute.org"
+          git commit -m "Auto-update Cromwell version to ${NEW_CROMWELL_HELM_V}"
+          git push https://broadbot:$BROADBOT_TOKEN@github.com/broadinstitute/leonardo.git ${NEW_BRANCH_NAME}
+          echo ::set-output name=NEW_BRANCH_NAME::${NEW_BRANCH_NAME}
+          echo ::set-output name=OLD_CROMWELL_HELM_V::${OLD_CROMWELL_HELM_V}
+          echo ::set-output name=NEW_CROMWELL_HELM_V::${NEW_CROMWELL_HELM_V}
+      
+      - name: Create leonardo PR
+        id: create-leo-pr
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ secrets.BROADBOT_TOKEN }}
+          script: |
+            const pull_response = await github.rest.pulls.create({
+              title: '$JIRA_ID Update Cromwell version to ${{ steps.update-and-push.outputs.NEW_CROMWELL_HELM_V }}',
```

Curly braces are "normally a good idea" to delimit the variable name, the colon is "what I tend to do in my own PRs". Take or leave both at your discretion:
```suggestion
              title: '${JIRA_ID}: Update Cromwell version to ${{ steps.update-and-push.outputs.NEW_CROMWELL_HELM_V }}',
```